python/qupled/classic.py

```diff
@@ -481,7 +484,68 @@ def plot(self, toPlot, matsubara : list[int] = None, rdfGrid : np.ndarray= None)
         super().plot(toPlot, matsubara, rdfGrid)
         if ("fxci" in toPlot):
             Plot.plot1D(self.scheme.freeEnergyGrid, self.scheme.freeEnergyIntegrand[1,:], "Coupling parameter", "Free energy integrand")
+
+class ESA(Stls):
+    """ Class to solve the hybrid ESA (Effective Static Approximation) scheme.
+
+    Class used to setup and solve the classical ESA scheme as described by
+    `Tobias Dornheim, Zhandos A. Moldabekov, and Panagiotis Tolias <https://journals.aps.org/prb/abstract/10.1103/PhysRevB.103.165102>`
+    This class inherits most of its methods and attributes from :obj:`~qupled.classic.Stls`
+
+    Args:
+        coupling: Coupling parameter.
+        degeneracy: Degeneracy parameter.  
+        chemicalPotential: Initial guess for the chemical potential, defaults to [-100.0, 100.0].
+        cutoff:  Cutoff for the wave-vector grid, defaults to 10.0.
+        matsubara: Number of matsubara frequencies, defaults to 128.
+        outputFrequency: Frequency used to print the recovery files, defaults to 10.
+        resolution: Resolution of the wave-vector grid, defaults to 0.1.
+    """
+
+    # Constructor
+    def __init__(self,
```

Comment:
Remove all the parameters that are not strictly necessary for the ESA scheme from the __init__ function